pkg/rtc/room.go

```diff
@@ -1191,11 +1190,22 @@ func (r *Room) pushAndDequeueUpdates(pi *livekit.ParticipantInfo, isImmediate bo
 		} else {
 			// different participant sessions
 			if existing.JoinedAt < pi.JoinedAt {
-				// existing is older, synthesize a DISCONNECT for older and
-				// send immediately along with newer session to signal switch
+				// existing is older, two scenarios to handle. In both cases,
+				// update is sent immediately as there is a new session.
+				// 1. Clients not using identity based reconnection:
+				//    synthesize a DISCONNECT for older session to ensure that
+				//    transition is explicit.
+				// 2. Clients that use identity to check for session changes:
+				//    these can handle session change without emitting events
+				//    that could be disruptive, i. e. emitting a `DISCONNECT`
+				//    state event followed by `ACTIVE` for same user could make
+				//    users of SDK to react to the transition in a disruptive way.
+				//    In this case, there is no need to synthesize `DISCONNECT`.
 				shouldSend = true
```

Comment:
Doing immediate send on session change even if not a publisher or `opts.immediate` is not set. I think that is good. Just wanted to highlight in case there are concerns about it. This case should be rare and updating immediately on a session change is good to ensure stale data is not living for too long.